services/src/main/java/org/apache/druid/cli/ServerRunnable.java

```diff
@@ -194,15 +202,40 @@ private DiscoverySideEffectsProvider(
       this.useLegacyAnnouncer = useLegacyAnnouncer;
     }
 
+    @VisibleForTesting
+    DiscoverySideEffectsProvider(
```

Comment:
Is it possible to make the private constructor delegate to this one?

Reply:
I'd need to refactor this to use Guice's AssistedInject Factory to avoid having 2 constructors - It's why I don't really like injected field members.

If you think it's worth the refactoring, I can make that change. I'll wait to see if there are any other comments before pushing up a new patch

Reply:
Since this is a release blocker for 0.19, I don't think refactoring is necessary. FWIW I don't really like AssistedInject, it's a bit too magical for my taste.

Reply:
If it's complicated, I'm ok with doing the refactoring in a follow up PR.

Reply:
I'll make this change in a follow up PR so we can push this through quickly